filters/block/block_test.go

```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
 
 		})
 	}
+}
+
+func TestMatcherErrorCases(t *testing.T) {
+	t.Run("maxBufferAbort", func(t *testing.T) {
+		r := &nonBlockingReader{initialContent: []byte("fppppppppp .class")}
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(r, toblockList, 5, maxBufferAbort)
+		p := make([]byte, len(r.initialContent))
+		_, err := bmb.Read(p)
+		if err != ErrMatcherBufferFull {
+			t.Errorf("Failed to get expected error %v, got: %v", ErrMatcherBufferFull, err)
+		}
+	})
+	t.Run("maxBuffer", func(t *testing.T) {
+		r := &nonBlockingReader{initialContent: []byte("fppppppppp .class")}
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(r, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(r.initialContent))
+		_, err := bmb.Read(p)
+		if err != nil {
+			t.Errorf("Failed to read: %v", err)
+		}
+	})
+
+	t.Run("maxBuffer read on closed reader", func(t *testing.T) {
+		pipeR, pipeW := io.Pipe()
+		initialContent := []byte("fppppppppp")
+		go pipeW.Write(initialContent)
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(pipeR, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(initialContent)+10)
+		pipeR.Close()
+		_, err := bmb.Read(p)
+		if err == nil || err.Error() != "io: read/write on closed pipe" {
+			t.Errorf("Failed to get correct read error: %v", err)
+		}
+	})
+
+	t.Run("maxBuffer read on initial closed reader", func(t *testing.T) {
+		pipeR, _ := io.Pipe()
+		initialContent := []byte("fppppppppp")
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(pipeR, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(initialContent)+10)
+		pipeR.Close()
+		bmb.Close()
+		_, err := bmb.Read(p)
+		if err == nil || err.Error() != "reader closed" {
+			t.Errorf("Failed to get correct read error: %v", err)
+		}
+	})
+}
+
+func TestBlockCreateFilterErrors(t *testing.T) {
+	spec := NewBlockFilter(1024)
+
+	t.Run("empty args", func(t *testing.T) {
+		args := []interface{}{}
+		if _, err := spec.CreateFilter(args); err == nil {
+			t.Error("CreateFilter with empty args should fail")
+		}
+	})
+
+	t.Run("non string args", func(t *testing.T) {
+		args := []interface{}{3}
+		if _, err := spec.CreateFilter(args); err == nil {
+			t.Error("CreateFilter with non string args should fail")
+		}
+	})
+}
+
+func TestBlock(t *testing.T) {
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte("OK"))
+	}))
```

Comment:
I believe you'd need to `defer backend.Close()`. Maybe you could add main_test.go like we did for other packages to catch this in the future

Reply:
fixed